packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -2701,6 +2702,161 @@ describe("typespec-client-generator-core: decorators", () => {
         code: "@azure-tools/typespec-client-generator-core/empty-client-name",
       });
     });
+
+    it("duplicate model client name with all language scopes", async () => {
+      const diagnostics = await runner.diagnose(
+      `
+      @service({
+        title: "Contoso Widget Manager",
+      })
+      namespace Contoso.WidgetManager;
+      
+      @error
+      model Error {
+        code: string;
+        message?: string;
+      }
+      
+      @clientName("Test")
+      model Widget {
+        @key
+        id: int32;
+      
+        description?: string;
+      }
+
+      model Test {
+        prop1: string;
+      }
+
+      @route("/test")
+      op test(): Test | Error;
+
+      op list(@query apiVersion: string): Widget[] | Error;
+      
+      @route("/widget/{id}")
+      op get(...Resource.KeysOf<Widget>): Widget | Error;
+      `
+      );
+
+      expectDiagnostics(diagnostics, [
+        {
+          code: "@azure-tools/typespec-client-generator-core/duplicate-model-name",
+        },
+        {
+          code: "@azure-tools/typespec-client-generator-core/duplicate-model-name",
+        }
+        ]
+      );
+    });
+    
+    it("duplicate enum client name with all language scopes", async () => {
+      const diagnostics = await runner.diagnose(
+      `
+      @service({
+        title: "Contoso Widget Manager",
+      })
+      namespace Contoso.WidgetManager;
+      
+      @error
+      model Error {
```

Comment:
I think a lot of those test have quite a few uncessary items, can it be simplified to just have the models needed, we don't need to make a full example service per test.

Reply:
simplified